lib/api/src/sys/imports.rs

```diff
@@ -111,6 +112,36 @@ impl Imports {
             .insert((ns.to_string(), name.to_string()), val.into());
     }
 
+    /// Imports (any) shared memory into the imports.
+    /// (if the module does not import memory then this function is ignored)
+    pub fn import_shared_memory(
+        &mut self,
+        module: &Module,
+        store: &mut impl AsStoreMut,
+    ) -> Option<VMSharedMemory> {
+        // Determine if shared memory needs to be created and imported
+        let shared_memory = module
+            .imports()
+            .memories()
+            .next()
+            .map(|a| *a.ty())
+            .map(|ty| {
+                let style = store.as_store_ref().tunables().memory_style(&ty);
+                VMSharedMemory::new(&ty, &style).unwrap()
+            });
+
+        if let Some(memory) = shared_memory {
+            self.define(
+                "env",
+                "memory",
+                Memory::new_from_existing(store, memory.clone().into()),
+            );
+            Some(memory)
+        } else {
+            None
+        }
+    }
```

Comment:
I don't think this should be in the public Wasmer API

Reply:
It should be addressed with commit [7cb4b0a](https://github.com/wasmerio/wasmer/pull/3153/commits/7cb4b0acd507fe1645f84f9637514cbf3c68257c)